backend/src/api/PollController.ts

```diff
@@ -0,0 +1,300 @@
+      }
+
+      const poll = await this.pollManager.vote(poll_id, userId, option_ids)
+
+      this.logger.info(`User #${userId} voted in poll #${poll_id}`, {
+        user_id: userId,
+        poll_id,
+        option_ids,
+      })
+
+      response.success({ poll })
+    } catch (err) {
+      if (err instanceof Error) {
+        if (err.message === 'Poll not found') {
+          return response.error('not-found', 'Poll not found', 404)
+        }
+        if (err.message === 'Poll has expired') {
+          return response.error('expired', 'Poll has expired', 403)
+        }
+        if (err.message === 'Multiple choice not allowed') {
+          return response.error('multiple-choice-not-allowed', 'Multiple choice not allowed', 400)
+        }
+        if (err.message === 'Invalid option ID') {
+          return response.error('invalid-option', 'Invalid option ID', 400)
+        }
+      }
+
+      this.logger.error('Vote error', { error: err, user_id: userId, poll_id })
+      return response.error('error', 'Unknown error', 500)
+    }
+  }
+
+  async getPoll(request: APIRequest<PollGetRequest>, response: APIResponse<PollGetResponse>) {
+    const userId = request.session.data.userId
+    const pollId = parseInt(request.params.pollId)
+
+    try {
+      const poll = await this.pollManager.getPoll(pollId, userId)
+      if (!poll) {
+        return response.error('not-found', 'Poll not found', 404)
+      }
+
+      response.success({ poll })
+    } catch (err) {
+      this.logger.error('Get poll error', { error: err, poll_id: pollId })
+      return response.error('error', 'Unknown error', 500)
+    }
+  }
+
+  async getPolls(request: APIRequest<PollListRequest>, response: APIResponse<PollListResponse>) {
+    const { site_id, limit, offset, active_only } = request.query
+
+    try {
+      const result = await this.pollManager.getPolls(
+        parseInt(site_id as string),
+        limit ? parseInt(limit as string) : undefined,
+        offset ? parseInt(offset as string) : undefined,
+        active_only === 'true',
+      )
+
+      response.success(result)
+    } catch (err) {
+      this.logger.error('Get polls error', { error: err, site_id })
+      return response.error('error', 'Unknown error', 500)
+    }
+  }
+
+  async getPollsBatch(request: APIRequest<PollBatchRequest>, response: APIResponse<PollBatchResponse>) {
+    const userId = request.session.data.userId
+    const { ids } = request.body
+
+    try {
+      const polls = await Promise.all(ids.map((id) => this.pollManager.getPoll(id, userId)))
+      const validPolls = polls.filter((poll): poll is PollEntity => poll !== null)
+
+      response.success({ polls: validPolls })
+    } catch (err) {
+      this.logger.error('Get polls batch error', { error: err, poll_ids: ids })
+      return response.error('error', 'Unknown error', 500)
+    }
+  }
+
+  async rescindVote(request: APIRequest<{ poll_id: number }>, response: APIResponse<PollVoteResponse>) {
```

Comment:
I won't comment about this method, assuming we're removing it to be replaced with `vote`